en/developer/appdev/articles/03-service-builder/04-business-logic-with-service-builder/02-creating-local-services/03-implementing-methods-to-get-and-count-entities.markdown

```diff
@@ -141,19 +143,21 @@ project's service methods: READ-WRITE and plain old READ. Specific service metho
 - `reindex`
 - `search`
 
-Otherwise, a READ-WRITE transaction is made. Methods that persist data (i.e.,
-add, update, or delete data) must execute in READ-WRITE transactions. Therefore,
-DO NOT use the words above to prefix names of data persistence methods. 
+Otherwise, the transactions executed are not read only. Methods that persist
+data (i.e., add, update, or delete data) must not use read-only transactions.
+Therefore, DO NOT use the words above to prefix names of data persistence
+methods. 
 
-Similarly, in methods recognized for READ transactions (methods whose names are
-prefixed with the words above), DO NOT call services that persist data. For
+Similarly, in methods identified for read-only transactions (methods whose names
+are prefixed with the words above), DO NOT call services that persist data. For
```

Comment:
@jhinkey the "DO NOT call services that persist data" is wrong, yes you can do that. That will start a new aop scope to allow modification. And actually this is the only right way to do it.

Therefore the following workaround "getOrAddMyEntity" to "addOrGetMyEntity" is really not necessary. We do "create" in "get" methods all the times ourselves. Like https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/service/impl/ClassNameLocalServiceImpl.java#L113

Reply:
@shuyangzhou I'm confused because the discussion in this forum post mentions that such a workaround was needed. 

https://liferay.dev/forums/-/message_boards/message/114748359

Reply:
That discuss is wrong, that "addUserBookmarksSet" call is a local method call, not a service method call, service method call goes through aop. All they need is to change it to "bookmarksSetLocalService.addUserBookmarksSet()" then it will work. That weird workaround can work, but definitely not the way we want to tell people to do, it should be avoid.